samples/csharp/sharedcontent/console/speech_synthesis_samples.cs

```diff
@@ -850,5 +850,93 @@ public static async Task SynthesisGetAvailableVoicesAsync()
                 }
             }
         }
+
+        private static async Task<SpeechSynthesisResult> SynthesizeOnceAsyncInternal(string key, string region,
+            string endpointId = null, string voiceName = null, string text = "Hello World!")
+        {
+            SpeechSynthesisResult synthesisResult = null;
+            var config = SpeechConfig.FromSubscription(key, region);
+            if (!string.IsNullOrEmpty(endpointId))
+            {
+                config.EndpointId = endpointId;
+            }
+
+            if (!string.IsNullOrEmpty(voiceName))
+            {
+                config.SpeechSynthesisVoiceName = voiceName;
+            }
+
+            // Creates a speech synthesizer using the default speaker as audio output.
+            using (var synthesizer = new SpeechSynthesizer(config))
+            {
+                synthesisResult = await synthesizer.SpeakTextAsync(text).ConfigureAwait(false);
+            }
+
+
+            return synthesisResult;
+        }
+
+        // Speech synthesis with backup subscription region.
+        public static async Task SynthesizeOnceToSpeakerAsyncSwitchSecondaryRegion()
+        {
+            // Create a speech resource with primary subscription key and service region.
+            // Also create a speech resource with secondary subscription key and service region
+            SpeechSynthesisResult synthesisResult = await SynthesizeOnceAsyncInternal("PrimarySubscriptionKey", "PrimarySubscriptionRegion" );
+            if (synthesisResult.Reason == ResultReason.Canceled)
+            {
+                SpeechSynthesisCancellationDetails details = SpeechSynthesisCancellationDetails.FromResult(synthesisResult);
+                if (details.ErrorCode == CancellationErrorCode.ConnectionFailure
```

Comment:
>ConnectionFailure [](http://example.com/codeflow?start=63&length=17)

The other two seem obvious, but is ConnectionFailure a little too generic to assume you should switch regions?

Reply:
Yes, connection failure is too generic. Some client issue causes connection failure. I think the sample is used in such scenario: the primary region worked yesterday, and don't work today without any code change. Then the failure reason is mostly a service issue